addons/mail/tests/test_discuss_controller.py

```diff
@@ -342,3 +344,31 @@ def test_mail_partner_from_email_authenticated(self):
             self.env["res.partner"].search_count([('email', '=', "john2@test.be")]),
             "'mail/message/post' does not create another user if there's already a user with matching email",
         )
+
+    def test_mail_cache_control_header(self):
+        testuser = self.env['res.users'].create({
+            'email': 'testuser@testuser.com',
+            'groups_id': [Command.set([self.ref('base.group_system')])],
+            'name': 'Test User',
+            'login': 'testuser',
+            'password': 'testuser',
+        })
+        test_user = self.authenticate("testuser", "testuser")
+        partner = self.env["res.users"].browse(test_user.uid).partner_id
+        self.channel.add_members(
+            self.env["res.users"].browse(test_user.uid).partner_id.ids
+        )
+        res = self.url_open(
+            url=f"/discuss/channel/{self.channel.id}/avatar_128?unique={self.channel._get_avatar_cache_key()}"
+        )
+        self.assertEqual(res.headers["Cache-Control"], f"public, max-age={STATIC_CACHE_LONG}")
+
+        res = self.url_open(
+            url=f"/discuss/channel/{self.channel.id}/partner/{partner.id}/avatar_128?unique={partner.write_date.isoformat()}"
+        )
+        self.assertEqual(res.headers["Cache-Control"], f"public, max-age={STATIC_CACHE_LONG}")
+
+        res = self.url_open(
+            url=f"/discuss/channel/{self.channel.id}/guest/{self.guest.id}/avatar_128?unique={self.guest.write_date.isoformat()}"
+        )
+        self.assertEqual(res.headers["Cache-Control"], f"public, max-age={STATIC_CACHE_LONG}")
```

Comment:
It would be nice to have the test that the cache is not set if unique is not provided too.